### PR TITLE
CMake: Add WITH_BOOST_STACKTRACE option to use libbacktrace

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  CMake: Add WITH_BOOST_STACKTRACE option to use libbacktrace

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

boost::stacktrace on linux showed addresses instead of functions/file/line

**Tests performed:**

Built, tested

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] There migth be issues when not linking jemalloc as boost::stacktrace wants "dl" library

### CODE REVIEW

**Comment by jackpoz** [comment]
Tested with gcc
image

Test with clang
image

**Comment by jackpoz** [comment]
CMake error if backtrace.h is not found:
image

File: dep/boost/CMakeLists.txt [hunk 1]
```diff
@@ -83,3 +83,23 @@ endif()
...
+
+if (WITH_BOOST_STACKTRACE)
+  if (BOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE)
+    CHECK_INCLUDE_FILE(${BOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE} HAS_BACKTRACE)
+  else()
+    CHECK_INCLUDE_FILE("backtrace.h" HAS_BACKTRACE)
+  endif()
+
+  if (NOT HAS_BACKTRACE)
+    message(FATAL_ERROR "Required header 'backtrace.h' not found. If building with a compiler other than GCC please specify the full path in the CMake option BOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE")
```
Comment (illfated) [inline dep/boost/CMakeLists.txt:95]: Punctuation details:
```suggestion
    message(FATAL_ERROR "Required header 'backtrace.h' not found. If building with a compiler other than GCC, please specify the full path in the CMake option BOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE.")
```
(You may skip the ending period dot, if you feel it takes away focus from the message.)

**Review by ghost — APPROVED** [review]
Looks quite good to me, especially since it is an optional addition to CMake.

**Comment by illfated** [comment]
Oh. Right. Good update. Forgot to notice which OS it supports and how it affects the PR code.

**Pull request merged**